Client/Frontend/Browser/FindInPageBar.swift

```diff
@@ -0,0 +1,136 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import UIKit
+
+protocol FindInPageBarDelegate: class {
+    func findInPage(findInPage: FindInPageBar, didTextChange text: String)
+    func findInPage(findInPage: FindInPageBar, didFindPreviousWithText text: String)
+    func findInPage(findInPage: FindInPageBar, didFindNextWithText text: String)
+    func findInPageDidPressClose(findInPage: FindInPageBar)
+}
+
+class FindInPageBar: UIView {
+    weak var delegate: FindInPageBarDelegate?
+    private let searchText = UITextField()
+    private let matchCountView = UILabel()
+    private let previousButton = UIButton()
+    private let nextButton = UIButton()
+
+    var currentResult = 0 {
+        didSet {
+            matchCountView.text = "\(currentResult)/\(totalResults)"
+        }
+    }
+
+    var totalResults = 0 {
+        didSet {
+            matchCountView.text = "\(currentResult)/\(totalResults)"
+            previousButton.enabled = totalResults > 1
+            nextButton.enabled = previousButton.enabled
+        }
+    }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        backgroundColor = UIColor.whiteColor()
+
+        searchText.addTarget(self, action: "didTextChange:", forControlEvents: UIControlEvents.EditingChanged)
```

Comment:
Yeah, it doesn't require them, it's just we arguably *should* prefix selectors with `@objc private` (I forgot the `private` here, thanks for the reminder!). For one, I think it's always a good practice to restrict method visibility as much as possible -- it'd be weird for a class to directly call another class's selector. Also, using `@objc private` can make a pretty clear indicator that it's a selector method as mentioned above (as long as we make a habit of using it consistently, anyway).